src/ui/components/SecondaryToolbox/TimelineContextAdapter.tsx

```diff
@@ -38,7 +38,7 @@ export default function TimelineContextAdapter({ children }: PropsWithChildren)
 
       // Pre-cache Pause data (required by legacy app code) before calling seek().
       // The new Console doesn't load this data but the old one requires it.
-      const cachedPauseId = getCachedPauseIdForExecutionPoint(executionPoint);
+      const cachedPauseId = getPauseIdForExecutionPointIfCached(executionPoint)?.value;
```

Comment:
Why does this have to call `.value`? That seems wrong. Is this a typo?

Reply:
This is correct - `getPauseIdForExecutionPointIfCached()` does not return the `pauseId` directly but rather an object containing it. Why? Because the cache may contain `undefined` values (*this* cache doesn't but the frame cache does, for example) and we need to distinguish between a cache hit with an `undefined` value and a cache miss.

Reply:
> the cache may contain `undefined` values

Interesting. Feels funny to me that `getValueIfCached` has a different return type/shape than `getValueSuspense` and `getValueAsync` although I see where you're coming from.

I'm curious– are there any caches where this distinction _does_ matter? or is it more of a conceptual one?

Either way, we can chat more about this later. Doesn't seem like anything to get hung up on for this PR 😄

Reply:
This is more of a conceptual thing. The FrameCache allows `undefined` values but the callers of `getFramesIfCached()` currently don't distinguish between a cache hit with an `undefined` value and a cache miss.

Reply:
Right, I understand that. A comparable API might be something like `Map` which has both a `has` and `get` method to differentiate these cases.